wnfs-common/src/blockstore/threadsafememoryblockstore.rs

```diff
@@ -0,0 +1,54 @@
+use crate::{BlockStore, BlockStoreError};
+use anyhow::Result;
+use async_trait::async_trait;
+use libipld::{Cid, IpldCodec};
+use std::{
+    borrow::Cow,
+    collections::HashMap,
+    sync::{Arc, RwLock},
+};
+
+/// A concurrency-friendly blockstore that you can mutate.
+#[derive(Debug, Default)]
+pub struct ThreadSafeMemoryBlockStore(Arc<RwLock<HashMap<String, Vec<u8>>>>);
+
+impl ThreadSafeMemoryBlockStore {
+    /// Creates a new in-memory block store.
+    pub fn new() -> Self {
+        Self::default()
+    }
+}
+
+impl Clone for ThreadSafeMemoryBlockStore {
+    fn clone(&self) -> Self {
+        Self::new()
+        //Self(RwLock::new(self.0.read().unwrap().clone()))
+    }
+}
```

Comment:
Nit: Don't you want to clone the `Arc` here? Also `#[derive(Clone)]` to make that happen.